template.yml

```diff
@@ -565,8 +567,6 @@ Resources:
       ManagedPolicyArns:
         # Transcode operations are always pulling FROM the artifact bucket
         - !Ref ArtifactBucketReadOnlyAccessPolicy
-        # The destination is a set of user-defined buckets
-        - !Ref DestinationBucketsWriteAccessPolicy
```

Comment:
What's the advantage of assuming this role, vs just granting it to the execution in the first place?

Reply:
Update per slack: ability to have a bunch of Porter functions with different permissions, but a single write-destination-S3 role for callers to grant to.

Reply:
Copying from the PR description: 

>In order to make this work, without having all of the state machine state resources (Lambda functions, Fargate tasks, etc), use a single, shared execution role that is also the role being granted acesss by the buckets, the resources maintain independent execution roles with properly-scoped policies, and the S3 operations that touch destination buckets assume the shared policy at run time, and use temporary credentials to make API calls to S3.

I agree this could potentially be overly complicated. I think it's easy to refactor if we hit that point though, and it wouldn't impact any buckets that are already set up to use the role.